crates/foundry-ui/src/components/typed.rs

```diff
@@ -0,0 +1,40 @@
+use console::Style;
+use serde::Serialize;
+
+use crate::Message;
+
+/// Generic textual message with `ty` prefix.
+///
+/// The type prefix can be stylized in text mode.
+/// e.g. "Tests: 1 passed, 1 failed"
+#[derive(Serialize)]
+pub struct TypedMessage<'a> {
```

Comment:
Typed is a bit misleading naming for me, it suggest it has something to do with the type definition itself or the type used.

Reply:
Hmm, agree, maybe "labeled" would be better, will rename it.